docs/coding-guidelines/adding-api-guidelines.md

```diff
@@ -43,13 +43,26 @@ should be added to `net10.0`. [More Information on TargetFrameworks](https://lea
 ## Documentation
 
 New public APIs must be documented with triple-slash comments on top of them. Visual Studio automatically generates the structure for you when you type `///`.
+See [guidelines for creating .NET API documentation](https://github.com/dotnet/dotnet-api-docs/wiki) for language and style. If your new API or the APIs it calls throw any exceptions, those need to be manually documented by adding the `<exception></exception>` elements.
 
-If your new API or the APIs it calls throw any exceptions, those need to be manually documented by adding the `<exception></exception>` elements.
+### Documentation Workflow
 
-After your change is merged, we will eventually port them to the dotnet-api-docs repo, where we will review them for language and proper style (For more information, see the [API writing guidelines](https://github.com/dotnet/dotnet-api-docs/wiki)).
+The documentation workflow depends on whether the assembly has the `UseCompilerGeneratedDocXmlFile` property set in its project file:
 
-Once the dotnet-api-docs change is merged, your comments will start showing up in the official API documentation at https://learn.microsoft.com, and later they'll appear in IntelliSense in Visual Studio and Visual Studio Code.
-Once the documentation is official, any subsequent updates to it must be made directly in https://github.com/dotnet/dotnet-api-docs/. It's fine to make updates to the triple slash comments later, they just won't automatically flow into the official docs.
+**For assemblies with `<UseCompilerGeneratedDocXmlFile>false</UseCompilerGeneratedDocXmlFile>`:**
+- The [dotnet-api-docs](https://github.com/dotnet/dotnet-api-docs) repo is the source of truth for documentation.
+- Triple-slash comments in source code are synced to dotnet-api-docs **only once** for newly introduced APIs.
+- After the initial sync, all subsequent documentation updates must be made directly in the dotnet-api-docs repo.
+- Older assemblies, including System.Private.CoreLib, typically follow this workflow.
+
+**For assemblies without this property (or with it set to `true`, which is the default):**
+- Source comments in this repo are the source of truth for documentation.
+- After your change is merged, we will eventually port them to the dotnet-api-docs repo, where we will review them for language and proper style (For more information, see the [API writing guidelines](https://github.com/dotnet/dotnet-api-docs/wiki)).
+- Once the dotnet-api-docs change is merged, your comments will start showing up in the official API documentation at https://learn.microsoft.com, and later they'll appear in IntelliSense in Visual Studio and Visual Studio Code.
+- Once the documentation is official, any subsequent updates to it must be made directly in https://github.com/dotnet/dotnet-api-docs/. It's fine to make updates to the triple slash comments later, they just won't automatically flow into the official docs.
```

Comment:
Line 62 seems to contradict Line 60. I understand 62 to say updates will not be merged to dotnet-api-docs and 60 says they will be merged. I'm guessing line 60 is the accurate one or there is still some subtlety I am missing.